test/session_tests/DocumentStoreTest.ts

```diff
@@ -182,6 +182,16 @@ describe('Document store test', () => {
       expect(product['@metadata']['Raven-Node-Type']).to.not.exist;  
       expect(product['@metadata']['@collection']).to.equals('@empty');      
     });
+
+    it('shouldn\'t set id on plain object without id attribute defined', async () => {
+      let product: Product = <Product>{name: "New Product"};    
+
+      session = store.openSession({requestExecutor});      
+      await session.store<Product>(product);
+      await session.saveChanges();
+
+      expect(product.id).to.be.empty;      
```

Comment:
Probably not a good idea, user can ignore it if they don't need it, and having the option to not have it is fine.
But just putting it by default looks better, I think.